metaflow/decorators.py

```diff
@@ -402,8 +405,11 @@ def _base_step_decorator(decotype, *args, **kwargs):
         if not hasattr(func, "is_step"):
             raise BadStepDecoratorException(decotype.name, func)
 
-        # Only the first decorator applies
-        if decotype.name in [deco.name for deco in func.decorators]:
+        # if `allow_multiple` is not `True` then only one decorator type is allowed per step
+        if (
+            decotype.name in [deco.name for deco in func.decorators]
+            and not decotype.allow_multiple
+        ):
```

Comment:
Minor nit - https://github.com/Netflix/metaflow/blob/273acb8c038a9ebacb444e89fd1a6b77689f00dd/metaflow/decorators.py#L64 should now read  `this decorator``